src/Lecture1.hs

```diff
@@ -98,7 +105,10 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str
+  | start <= 0 = take (end + 1) str
+  | otherwise = subString (start - 1) (end - 1) (tail str)
```

Comment:
This looks like a correct implementation 👍🏻 
Alternatively, you can use standard functions `drop` and `take` to avoid implementing manual recursion here 🙂